chromadb/embedding_functions/__init__.py

```diff
@@ -0,0 +1,174 @@
+def register_embedding_function(ef_class: Type[EmbeddingFunction[Embeddable]]) -> None:
+    """Register a custom embedding function.
+
+    Args:
+        ef_class: The embedding function class to register.
+    """
+    # Create an instance to get the name
+    ef_instance = ef_class()
+    name = ef_instance.name()
+    supported_embedding_functions[name] = ef_class
+
+
+# Function to convert config to embedding function
+def config_to_embedding_function(
+    config: Dict[str, Any]
+) -> EmbeddingFunction[Documents]:
+    """Convert a config dictionary to an embedding function.
+
+    Args:
+        config: The config dictionary.
+
+    Returns:
+        The embedding function.
+    """
+    if "name" not in config:
+        raise ValueError("Config must contain a 'name' field.")
+
+    name = config["name"]
+    if name not in supported_embedding_functions:
+        raise ValueError(f"Unsupported embedding function: {name}")
+
+    ef_config = config.get("config", {})
+
+    # Handle each embedding function type specifically
+    if name == "cohere":
+        return CohereEmbeddingFunction.build_from_config(ef_config)
+    elif name == "openai":
+        return OpenAIEmbeddingFunction.build_from_config(ef_config)
+    elif name == "huggingface":
+        return HuggingFaceEmbeddingFunction.build_from_config(ef_config)
+    elif name == "huggingface_server":
+        return HuggingFaceEmbeddingServer.build_from_config(ef_config)
+    elif name == "sentence_transformer":
+        return SentenceTransformerEmbeddingFunction.build_from_config(ef_config)
+    elif name == "google_palm":
+        return GooglePalmEmbeddingFunction.build_from_config(ef_config)
+    elif name == "google_generative_ai":
+        return GoogleGenerativeAiEmbeddingFunction.build_from_config(ef_config)
+    elif name == "google_vertex":
+        return GoogleVertexEmbeddingFunction.build_from_config(ef_config)
+    elif name == "ollama":
+        return OllamaEmbeddingFunction.build_from_config(ef_config)
+    elif name == "instructor":
+        return InstructorEmbeddingFunction.build_from_config(ef_config)
+    elif name == "jina":
+        return JinaEmbeddingFunction.build_from_config(ef_config)
+    elif name == "voyageai":
+        return VoyageAIEmbeddingFunction.build_from_config(ef_config)
+    elif name == "onnx_mini_lm_l6_v2":
+        return ONNXMiniLM_L6_V2.build_from_config(ef_config)
+    elif name == "open_clip":
+        return OpenCLIPEmbeddingFunction.build_from_config(ef_config)
+    elif name == "roboflow":
+        return RoboflowEmbeddingFunction.build_from_config(ef_config)
+    elif name == "text2vec":
+        return Text2VecEmbeddingFunction.build_from_config(ef_config)
+    elif name == "amazon_bedrock":
+        return AmazonBedrockEmbeddingFunction.build_from_config(ef_config)
+    elif name == "chroma_langchain":
+        return ChromaLangchainEmbeddingFunction.build_from_config(ef_config)
```

Comment:
could we just look up `supported_embedding_functions` instead of manually having this switch?